src/ui/pages/RevealPrivateKey/RevealPrivateKey.tsx

```diff
@@ -0,0 +1,171 @@
+            }}
+          >
+            <UIText kind={'body/regular'} style={{ wordBreak: 'break-word' }}>
+              {privateKey}
+            </UIText>
+          </Surface>
+        </BlurredToggle>
+        <div style={{ textAlign: 'center' }}>
+          <Button
+            kind="regular"
+            size={36}
+            type="button"
+            onClick={handleCopy}
+            style={{ paddingLeft: 16, paddingRight: 16 }}
+          >
+            <HStack gap={8}>
+              {React.createElement(isCopySuccess ? CheckIcon : CopyIcon, {
+                style: { display: 'block', width: 20, height: 20 },
+              })}
+              <ZStack
+                hideLowerElements={true}
+                justifyContent="start"
+                style={{ textAlign: 'left' }}
+              >
+                {isCopySuccess ? <span>Copied to Clipboard</span> : null}
+                <span aria-hidden={isCopySuccess}>Copy to Clipboard</span>
+              </ZStack>
+            </HStack>
+          </Button>
+        </div>
+      </VStack>
+      <Button
+        style={{ marginTop: 'auto' }}
+        autoFocus={true}
+        onClick={() => onSubmit()}
+      >
+        Done
+      </Button>
+      <PageBottom />
+    </PageColumn>
+  );
+}
+
+export function RevealPrivateKey() {
+  useBackgroundKind(whiteBackgroundKind);
+
+  const [params, setSearchParams] = useSearchParams();
+  const groupId = params.get('groupId');
+  const address = params.get('address');
+  invariant(groupId, 'groupId param is required for RevealPrivateKey view');
+  invariant(address, 'address param is required for RevealPrivateKey view');
+
+  const updateSearchParam = (
+    key: string,
+    value: string,
+    navigateOptions?: Parameters<typeof setSearchParams>[1]
+  ) => {
+    const newParams = new URLSearchParams(params);
+    newParams.set(key, value);
+    setSearchParams(newParams, navigateOptions);
+  };
+
+  const navigate = useNavigate();
+
+  const { data: walletGroup, isLoading } = useWalletGroup({ groupId });
+  if (isLoading || !walletGroup) {
+    return null;
+  }
+  assertSignerContainer(walletGroup.walletContainer);
+
+  return (
+    <>
+      {params.has('step') == false ? (
+        <PageColumn>
+          <NavigationTitle title="Private Key" />
+          <PageTop />
+          <VerifyUser
+            text={'Verification is required to show your secret key'}
+            onSuccess={() =>
+              updateSearchParam('step', 'revealSecret', { replace: true })
+            }
+          />
+          <PageBottom />
+        </PageColumn>
+      ) : null}
+      {params.get('step') === 'revealSecret' ? (
+        <Reveal
+          groupId={groupId}
+          address={address}
+          onSessionExpired={() => {
+            updateSearchParam('step', 'verifyUser', { replace: true });
+          }}
+          onSubmit={() => {
+            navigate('/');
```

Comment:
I think, we should zeroize here, shouldn't we?

Reply:
The `zeroizeAfterSubmission` function does the following:
```
export function zeroizeAfterSubmission() {
  /** Call this function to remove sensitive input values from runtime memory */
  const inputs = document.querySelectorAll('input[type=password]');
  inputs.forEach((input) => {
    if (input instanceof HTMLInputElement) {
      input.value = '';
    }
  });
  queryClient.getMutationCache().clear();
}
```
So, it specifically targets inputs with `type=password` and clears the mutation cache. But from what I see, we don't have any input `type="password"` on this screen, and there are no mutations either